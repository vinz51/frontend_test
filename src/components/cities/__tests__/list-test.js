import City from "api/models/city";
import CityItem from "../item";
import CitiesList from "../list";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import React from "react";

describe("CitiesList", () => {
  const cities = [
    new City({
      city: "Paris",
      rank: "1",
      state: "Ile de France",
    }),
    new City({
      city: "Reims",
      rank: "42",
      state: "Marne",
    }),
  ];

  describe("Snaphosts", () => {
    function getMyTreeComponent(props = {}) {
      return renderer.create(<CitiesList {...props} />).toJSON();
    }

    it("renders", () => {
      expect(getMyTreeComponent()).toMatchSnapshot();
    });

    it("renders with cities", () => {
      expect(getMyTreeComponent({ cities })).toMatchSnapshot();
    });
  });

  describe("Actions", () => {
    it("clicks on the cityItem", () => {
      const myOnCityClick = jest.fn();
      const myComponent = mount(
        <CitiesList cities={cities} onCityClick={myOnCityClick} />
      );

      expect(myComponent.find(CityItem).at(0).prop("active")).toBe(false);
      expect(myComponent.find(CityItem).at(1).prop("active")).toBe(false);

      // triggers the onClick on the item
      myComponent.find(CityItem).at(0).prop("onClick")();

      expect(myOnCityClick).toHaveBeenNthCalledWith(1, cities[0]);

      // Update component to set active city
      myComponent.setProps({ city: cities[0] });
      myComponent.update();

      // Expect that the clicked city is well activated
      expect(myComponent.find(CityItem).at(0).prop("active")).toBe(true);
      expect(myComponent.find(CityItem).at(1).prop("active")).toBe(false);
    });
  });
});
