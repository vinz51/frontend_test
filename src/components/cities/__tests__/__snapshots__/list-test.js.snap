// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList Snaphosts renders 1`] = `
Array [
  <ul
    className="sc-Axmtr iHRTWZ"
  />,
  <button
    className="sc-AxgMl ekIFmV sc-AxheI jkaUqU"
    disabled={true}
    onClick={[Function]}
  >
    See more
  </button>,
]
`;

exports[`CitiesList Snaphosts renders with cities 1`] = `
Array [
  <ul
    className="sc-Axmtr iHRTWZ"
  />,
  <button
    className="sc-AxgMl gQAZnk sc-AxheI jkaUqU"
    disabled={false}
    onClick={[Function]}
  >
    See more
    <span
      className="sc-AxmLO iGeVhH"
    >
      0
      /
      2
    </span>
  </button>,
]
`;
